File: TP3/JAVA/src/main/resources/EtudiantBoursier.java
package main.resources;

import java.util.List;

import src.Cours;
import src.Etudiant;

public class EtudiantBoursier extends Etudiant {
    
    public EtudiantBoursier(String nom, int age, String numeroEtudiant, double moyenne, List<Cours> cours) {
        super(nom, age, numeroEtudiant, moyenne, cours);
    }

    @Override
    public void afficherDetails() {
        System.out.println("Etudiant: " + getNom()
                + ", Age: " + getAge()
                + ", Moyenne: " + this.getMoyenne()
                + ", Cet etudiant est boursier.");
    }
}

File: TP3/PYTHON/src/pattern/Decorator/EtudiantBoursier.py
from src.pattern.Decorator.EtudiantDecorator import EtudiantDecorator

class EtudiantBoursier(EtudiantDecorator):
    def afficher_statut(self):
        nom = self._etudiant.get_nom()
        return f"{nom} est un étudiant boursier, il est exonéré de frais."